stingray/io.py

```diff
@@ -498,14 +501,152 @@ def _retrieve_ascii_object(filename, **kwargs):
+
+    colsassign: dictionary iterable
+        This indicates the correct tables to which to assign columns
+        to. If this is None or if a column is not provided, it/they will
+        be assigned to the first table.
+
+        For example, [{'gti':'GTI'}] indicates that gti values should be 
+        stored in GTI table.
+    """
+    tables = []
+
+    if 'colsassign' in list(kwargs.keys()):
+        colsassign = kwargs['colsassign']
+        iscolsassigned = True
+    else:
+        iscolsassigned = False
+
+    if 'tnames' in list(kwargs.keys()): 
+        tables = kwargs['tnames']
+    else:
+        tables = ['MAIN']
+    
+    items = vars(object)
+    attrs = [name for name in items if items[name] is not None]
+    
+    cols = []
+    hdrs = []
+
+    for t in tables:
+        cols.append([])
+        hdrs.append(fits.Header())
+    
+    for attr in attrs:
+        data = items[attr]
+
+        if iscolsassigned and attr in colsassign.keys():
+            index = tables.index(colsassign[attr])
+        else:
+            index = 0
+        
+        # If data is a single number, store as metadata.
+        if _isattribute(data):   
+            hdrs[index][attr] = data
+        # If data is an array or list, insert as table column.
+        else:
+            try:
+                cols[index].append(fits.Column(name=attr,format=_lookup_format(data[0]), 
+                    array=data))
+            except IndexError:
+                # To account for numpy arrays of type 'None' (0-d)
+                pass
+
+    tbhdu = fits.HDUList()
+
+    # Create binary tables
+    for i in range(0, len(tables)):
+        if cols[i] != []:
+            tbhdu.append(fits.BinTableHDU.from_columns(cols[i], header=hdrs[i], name=tables[i]))
+    
+    tbhdu.writeto(filename)
+
+def _retrieve_fits_object(filename, **kwargs):
+    """
+    Retrieves a fits format class object.
+
+    Parameters
+    ----------
+    filename: str
+        The name of file with which object was saved
+
+    Additional Keyword Parameters
+    -----------------------------
+    cols: str iterable
+        The names of columns to extract from fits tables.
+
+    Returns
+    -------
+    data: dictionary
+        Loads the data from a fits object file and returns
+        in dictionary format.
+    """
+    data = {}
+
+    if 'cols' in list(kwargs.keys()):
+        cols = kwargs['cols']
+    else:
+        cols = []
+
+    with fits.open(filename) as hdulist:
+        fits_cols = []
+
+        for i in range(1,len(hdulist)):
+            fits_cols.append([h.lower() for h in hdulist[i].data.names])
+
+        for c in cols:
+            for i in range(0, len(fits_cols)):
+                hdr_keys = [h.lower() for h in hdulist[i+1].header.keys()]
+
+                if c in fits_cols[i]:
+                    data[c] = hdulist[i+1].data[c]
+                elif c.lower() in hdr_keys:
+                    data[c] = hdulist[i+1].header[c]
+    
+    return data
+
```

Comment:
`load_events_and_gtis` is not really a general function. It does what the name says (retrieving event times and GTIs) and it's easy for the user to just use it. For one thing, it hides the fact of having different missions using different names for the `GTI` extension (e.g. RXTE using `STDGTI`, others using `GTI`). This new method is too general and it's on the user to know the format of data. The way to obsolete `load_events_and_gtis` would be to do the same intuitive operation using the new infrastructure in a better way. It's surely something to work on, eventually. Maybe using a more clever database-like (or config file) approach to tell the name of extensions in different missions instead of using the current if-then-else approach that is not exactly elegant. 
Just try not to delete methods before their functionality has been fully replaced.